src/Message/PurchaseRequest.php

```diff
@@ -72,7 +72,8 @@ class PurchaseRequest extends AuthorizeRequest
 
     public function getData()
     {
-        if ($this->parameters->get('transactionReference')) {
+        $realCardExists = ($this->getCard() || $this->getCardReference());
+        if ($this->parameters->get('transactionReference') && !$realCardExists) {
```

Comment:
@lukeholder simple enough. I'm curious how the expectation here failed. 

The original condition is meant to differentiate between a case of a reference sale or card sale. According to this change, checking for card data is needed to truly make that determination. Seems odd, so I'm curious what you're using the transaction reference field for in a card sale...

 In all other cases throughout the API (eg. voiding, etc) transaction reference === the `ORIGID`.  Can you show me what you're doing with this field?